src/Symfony/Component/Serializer/Tests/SerializerTest.php

```diff
@@ -706,3 +784,21 @@ interface NormalizerAwareNormalizer extends NormalizerInterface, NormalizerAware
 interface DenormalizerAwareDenormalizer extends DenormalizerInterface, DenormalizerAwareInterface
 {
 }
+
+class Dto
+{
+    /**
+     * @var \DateTimeImmutable
```

Comment:
Change it to ` @var int` and exceptions won't be collected.

`Symfony\Component\Serializer\Exception\NotNormalizableValueException : The type of the "foo" attribute for class "Symfony\Component\Serializer\Tests\Dto" must be one of "int" ("string" given).`

Reply:
That's expected. The point of this feature is to rely on the objets being denormalized to validate inputs. If those objects don't do that, validation should be implemented in custom denormalizers like we have here with `DateTimeNormalizer`. In the case of having `@var int`, we would have none, hence no validation error.

Reply:
That is a strange a bit. A common use deserialization in an API is to convert a json request to a DTO, which this solution is not suitable. It won't collect a lot of type errors in the request.

Reply:
Actually that may be handled thanks to #35235. I probably need to update more denormalizers.